phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java

```diff
@@ -2006,4 +2007,17 @@ public void run() {
         thread.setDaemon(true);
         thread.start();
     }
+
+    /**
+     * Confirms that no storeFile has refCount leakage
+     *
+     * @throws IOException if refCount is leaked
+     */
+    protected synchronized static void confirmStoreRefCountLeak()
```

Comment:
I think that throwing an exception here may interfere with the cluster shutdown logic that this is called from.

Either catch and log it (and fail the test), or make sure that it doesn't preclude cluster shutdown every invocation point.

Reply:
I think better catch and log it and also throw AssertionError with explanation.

Reply:
Throwing any exception would also stop the minicluster shutdown process, unless you catch it at the caller.

We should make sure that shutdownMiniCluster() will get called even if confirmStoreRefCountLeak() fails.

```
@AfterClass
public static void shutdown() {
  confirmStoreRefCountLeak();
  shutdownMiniCluster();
}
```

Reply:
Updated PR should work

Reply:
Let me update 4.x PR in the meanwhile.